src/awkward/operations/ak_from_buffers.py

```diff
@@ -157,11 +159,28 @@ def getkey(form, attribute):
 
 
 def _from_buffer(nplike, buffer, dtype, count, byteorder):
-    array = nplike.frombuffer(buffer, dtype=dtype, count=count)
-    if byteorder != ak._util.native_byteorder:
-        return array.byteswap(inplace=False)
-    else:
+    if nplike.is_own_array(buffer):
+        array = buffer.view(dtype)
+
+        if array.ndim != 1:
+            raise ak._errors.wrap_error(
+                TypeError(f"dimensionality of array should be 1, not ({array.ndim})")
+            )
+        # Require 1D
```

Comment:
Does `ak.to_buffers` ever return a multidimensional array? For instance, from a `NumpyArray` node? We want

```python
ak.from_buffers(*ak.to_buffers(x)) == x
```

to continue to be true.

Reply:
I was getting a bit caught up with not breaking existing ingest, yet I didn't even think about round-tripping. I've fixed this, and ensured that we don't copy any data.